pkg/api/v1alpha1/tinkerbelltemplateconfig_test.go

```diff
@@ -0,0 +1,183 @@
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			testName: "multiple tinkerbell template configs",
+			fileName: "testdata/cluster_1_21_valid_multiple_tinkerbell_templates.yaml",
+			wantConfigs: map[string]*TinkerbellTemplateConfig{
+				"tink-test-1": {
+					TypeMeta: metav1.TypeMeta{
+						Kind:       TinkerbellTemplateConfigKind,
+						APIVersion: SchemeBuilder.GroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "tink-test-1",
+					},
+					Spec: TinkerbellTemplateConfigSpec{
+						Template: tinkerbell.Workflow{
+							Version:       "0.1",
+							Name:          "tink-test-1",
+							GlobalTimeout: 6000,
+							ID:            "",
+							Tasks: []tinkerbell.Task{
+								{
+									Name:       "tink-test-1",
+									WorkerAddr: "{{.device_1}}",
+									Volumes: []string{
+										"/dev:/dev",
+										"/dev/console:/dev/console",
+										"/lib/firmware:/lib/firmware:ro",
+									},
+									Actions: []tinkerbell.Action{
+										{
+											Name:    "stream-image",
+											Image:   "image2disk:v1.0.0",
+											Timeout: 360,
+											Environment: map[string]string{
+												"IMG_URL":    "",
+												"DEST_DISK":  "/dev/sda",
+												"COMPRESSED": "true",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				"tink-test-2": {
+					TypeMeta: metav1.TypeMeta{
+						Kind:       TinkerbellTemplateConfigKind,
+						APIVersion: SchemeBuilder.GroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "tink-test-2",
+					},
+					Spec: TinkerbellTemplateConfigSpec{
+						Template: tinkerbell.Workflow{
+							Version:       "0.1",
+							Name:          "tink-test-2",
+							GlobalTimeout: 6000,
+							ID:            "",
+							Tasks: []tinkerbell.Task{
+								{
+									Name:       "tink-test-2",
+									WorkerAddr: "{{.device_1}}",
+									Volumes: []string{
+										"/dev:/dev",
+										"/dev/console:/dev/console",
+										"/lib/firmware:/lib/firmware:ro",
+									},
+									Actions: []tinkerbell.Action{
+										{
+											Name:    "stream-image",
+											Image:   "image2disk:v1.0.0",
+											Timeout: 360,
+											Environment: map[string]string{
+												"IMG_URL":    "",
+												"DEST_DISK":  "/dev/sda",
+												"COMPRESSED": "true",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			got, err := GetTinkerbellTemplateConfig(tt.fileName)
+			if (err != nil) != tt.wantErr {
+				t.Fatalf("GetTinkerbellTemplateConfig() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if !reflect.DeepEqual(got, tt.wantConfigs) {
+				t.Fatalf("GetTinkerbellTemplateConfig() = %#v, want %#v", got, tt.wantConfigs)
+			}
```

Comment:
Suggest the use of an assertion library. Assertion libraries really help document intent in a succinct way. 

```suggestion
            g := gomega.NewWithT(t)
			got, err := GetTinkerbellTemplateConfig(tt.fileName)
			g.Expect(err).To(gomega.BeEquivalentTo(tt.wantErr))
			g.Expect(gpt).To(gomega.BeEquivalentTo(tt.wanrConfigs))
```

Reply:
Sure! 
I just followed some tests we had defined earlier. I can change it to use asserts.